src/components/application_manager/src/application_manager_impl.cc

```diff
@@ -925,6 +925,33 @@ void ApplicationManagerImpl::CreatePendingApplication(
   SendUpdateAppList();
 }
 
+void ApplicationManagerImpl::SetPendingApplicationState(
+    const transport_manager::ConnectionUID connection_id,
+    const transport_manager::DeviceInfo& device_info) {
+  std::string name = device_info.name();
+  auto it = pending_device_map_.find(name);
+  if (it == pending_device_map_.end()) {
```

Comment:
Will need to use the `pending_device_map_lock_` here. Need to rebase on or merge`cloud_app_connection_status` branch.